tools/chunk/src/main/scala/org/allenai/nlpstack/chunk/OpenNlpChunker.scala

```diff
@@ -6,18 +6,24 @@ import org.allenai.nlpstack.core.{ ChunkedToken, Chunker, PostaggedToken }
 import opennlp.tools.chunker.{ ChunkerME, ChunkerModel }
 
 class OpenNlpChunker extends Chunker {
-  private val chunker = new ChunkerME(OpenNlpChunker.model)
+  //Added ThreadLocal to prevent concurrency issues
+  private final val chunker: ThreadLocal[ChunkerME] = new ThreadLocal[ChunkerME]() {
+    override protected def initialValue(): ChunkerME = new ChunkerME(OpenNlpChunker.model)
+  }
 
-  def chunkPostagged(tokens: Seq[PostaggedToken]): Seq[ChunkedToken] = {
+  def chunkPostaggedOriginal(tokens: Seq[PostaggedToken]): Seq[ChunkedToken] = {
     // OpenNLP uses : as the postag for hyphens, but we use HYPH, so we change it back before
     // sending it to the chunker.
     val fixedTokens = tokens.map { t =>
       if (t.string == "-") PostaggedToken(t, ":") else t
     }
 
-    val chunks = chunker.chunk(tokens.map(_.string).toArray, fixedTokens.map(_.postag).toArray)
+    val chunks = chunker.get().chunk(tokens.map(_.string).toArray, fixedTokens.map(_.postag)
+      .toArray)
     (tokens zip chunks) map { case (token, chunk) => ChunkedToken(token, chunk) }
   }
+
+  def chunkPostagged(tokens: Seq[PostaggedToken]): Seq[ChunkedToken] = chunkPostaggedOriginal(tokens)
```

Comment:
There is an extra level of indirection here we don't need anymore. The work can be done directly in `chunkPostagged`.